core/helpers/hardhat/slotOverwrite.ts

```diff
@@ -0,0 +1,116 @@
+import hre from 'hardhat';
+import { ethers } from 'ethers';
+import { getContractAddressByName } from '../../src/contracts';
+import BigNumber from '../../src/bignumber';
+import { EthereumProvider } from 'hardhat/types';
+import { formatToHex } from '../format';
+import { pad32, concat, stripZeros } from '../hex';
+import { TEST_NETWORK, HARDHAT_PUBLIC_KEY } from '../../helpers/constants';
+
+export const generateMappingSlotAddress = (
+    mappingStartSlot: string,
+    key: string,
+    languageFormat: 'vyper' | 'solidity' = 'solidity'
+) => {
+    if (languageFormat === 'solidity') {
+        return stripZeros(ethers.utils.keccak256(concat(pad32(key), pad32(mappingStartSlot))));
+    }
+    return stripZeros(ethers.utils.keccak256(concat(pad32(mappingStartSlot), pad32(key))));
+};
+
+export const overwriteUintValueInAddress = async (
+    address: string,
+    slotAddress: string,
+    newValue: BigNumber,
+    provider: EthereumProvider = hre.network.provider
+) => {
+    const hexValue = formatToHex(newValue, 32);
+    const storageToWrite = [address, slotAddress, hexValue];
+    await provider.send('hardhat_setStorageAt', storageToWrite);
+};
+
+export const overwriteUintValue = async (
+    contractName: string,
+    slotAddress: string,
+    newValue: BigNumber,
+    provider: EthereumProvider = hre.network.provider
+) => {
+    const contractAddress = await getContractAddressByName(TEST_NETWORK, contractName);
+    await overwriteUintValueInAddress(contractAddress, slotAddress, newValue, provider);
+};
+
+export const overwriteUintMapping = async (
+    contractName: string,
+    mappingSlotAddress: string,
+    mappingKey: string,
+    newValue: BigNumber,
+    provider: EthereumProvider = hre.network.provider
+) => {
+    const slotAddress = generateMappingSlotAddress(mappingSlotAddress, mappingKey);
+    await overwriteUintValue(contractName, slotAddress, newValue, provider);
+};
+
+export const overwriteUintMappingInAddress = async (
+    contractAddress: string,
+    mappingSlotAddress: string,
+    mappingKey: string,
+    newValue: BigNumber,
+    provider: EthereumProvider = hre.network.provider,
+    languageFormat: 'vyper' | 'solidity' = 'solidity'
+) => {
+    const slotAddress = generateMappingSlotAddress(mappingSlotAddress, mappingKey, languageFormat);
+    await overwriteUintValueInAddress(contractAddress, slotAddress, newValue, provider);
+};
+
+export const overwriteUintTable = async (
+    contractName: string,
+    mappingSlotAddress: string,
+    tableRowKey: string,
+    tableColumnKey: string,
+    newValue: BigNumber,
+    provider: EthereumProvider = hre.network.provider
+) => {
+    const rowAddress = generateMappingSlotAddress(mappingSlotAddress, tableRowKey);
+    await overwriteUintMapping(contractName, rowAddress, tableColumnKey, newValue, provider);
+};
+
+export const runSlotDiscoveryLoop = async (
```

Comment:
- [ ] Please make clear from the function name what it receives and what it outputs/modifies
- [ ] New: Let's move erc20-related helpers into a separate file?

Reply:
There's not a big enough cluster of helpers that are strictly designed to communicate with erc20 tokens (i found 1) only. so i would stick to the current distinction.